jax/_src/lax/linalg.py

```diff
@@ -681,7 +691,7 @@ def _triangular_solve_gpu_translation_rule(
     a = xops.Conj(a)
     conjugate_a = False
   if batch > 1 and m <= 32 and n <= 32:
```

Comment:
@hawkinsp So we are only using the cusolver.trsm for small matrices when they are batched, is that correct?

Reply:
Yes. XLA lowers TRSM to the unbatched kernel always; this is just a small refinement over what XLA itself does:
https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/xla/service/gpu/triangular_solve_thunk.cc;l=89

The right end state here would be to delete this code and have XLA emit the batched triangular solve where applicable. I just never quite got to that!

Reply:
And if the matrices are large it does not matter that XLA emits the unbatched version, since the performance is the same ?

Reply:
Yes, that's the intuition. It's possible the heuristic could use some tuning.

Reply:
Alright. Thanks for the clarification!